gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitMonitoringService.java

```diff
@@ -307,7 +369,10 @@ void moveCheckpointAndHashesForward() throws IOException {
       ObjectId oldHeadTree = git.getRepository().resolve(this.lastProcessedGitHash + "^{tree}");
 
       // refresh to latest and reset hard to handle forced pushes
-      this.git.fetch().setRemote(REMOTE_NAME).call();
+      FetchCommand fetchCommand = this.git.fetch().setRemote(REMOTE_NAME);
+      fetchCommand = (this.providerSessionFactoryEither instanceof Either.Left) ? fetchCommand.setCredentialsProvider(getCredentialsProvider())
```

Comment:
I think you can pass null for the parameters to avoid splitting the call here.

Reply:
Thanks! Addressed the comment.